commands/src/main/java/org/wildfly/extras/creaper/commands/security/realms/AddJaasAuthentication.java

```diff
@@ -0,0 +1,92 @@
+package org.wildfly.extras.creaper.commands.security.realms;
+
+import org.wildfly.extras.creaper.commands.foundation.offline.xml.GroovyXmlTransform;
+import org.wildfly.extras.creaper.commands.foundation.offline.xml.Subtree;
+import org.wildfly.extras.creaper.core.ServerVersion;
+import org.wildfly.extras.creaper.core.offline.OfflineCommandContext;
+import org.wildfly.extras.creaper.core.online.OnlineCommandContext;
+import org.wildfly.extras.creaper.core.online.operations.Address;
+import org.wildfly.extras.creaper.core.online.operations.Operations;
+import org.wildfly.extras.creaper.core.online.operations.Values;
+import org.wildfly.extras.creaper.core.online.operations.admin.Administration;
+
+/**
+ * Add jaas authentication to security realm.
+ */
+public final class AddJaasAuthentication extends AbstractAddSecurityRealmSubElement {
+
+    private final String name;
+    private final Boolean assignGroups;
+
+    private AddJaasAuthentication(Builder builder) {
+        super(builder);
+        this.name = builder.name;
+        this.assignGroups = builder.assignGroups;
+    }
+
+    @Override
+    public void apply(OnlineCommandContext ctx) throws Exception {
+        if (assignGroups != null) {
+            if (ctx.version.lessThan(ServerVersion.VERSION_3_0_0)
+                    && !ctx.version.equalTo(ServerVersion.VERSION_1_7_0)) {
```

Comment:
Sorry, I forgot about one thing! This will no longer be enough, there's `VERSION_1_8_0` now. I think a better condition would be: `ctx.version.lessThan(ServerVersion.VERSION_1_7_0) || ctx.version.inRange(ServerVersion.VERSION_2_0_0, ServerVersion.VERSION_2_2_0)`, but feel free to write the condition on your own.